repo2docker/utils.py

```diff
@@ -469,14 +469,33 @@ def is_local_pip_requirement(line):
     if line.startswith(("--requirement", "--constraint")):
         # as above but flags are spelt out
         return True
+    # the `--pre` flag is a global flag and should appear on a line by itself
+    # we just care that this isn't a "local pip requirement"
+    if line.startswith("--pre"):
+        return False
+
     # strip off things like `--editable=`. Long form arguments require a =
     if line.startswith("--"):
-        line = line.split("=", 1)[1]
+        # if splitting doesn't work it is probably because the line contains
+        # a syntax error or our "parser" is too simplistic so we return True
+        # just in case
```

Comment:
Is there a nicer way to handle the above comment so that it is simultaneously close to the code it is about and not repeated a few lines later?

Reply:
Yeah, that works. Or use:

Or for this one, we can check for presence of `=` sign before splitting, so we know that there will be two items:

```python
if line.startswith("--") and '=' in line:
    _, line = line.split("=", 1)
```

and handle length 1 or 2 in the later split:

```python
if line.startswith("-"):
    _, *rest = line.split(None, 1)
    if not rest:
        # no argument after `--flag`, skip line
        return False
    line = rest[0]
```

...or regular expressions!

```python
# pattern to split lines that start with one or two leading `-`
# takes 'argument s' off of:
# --flag=argument s
# -f argument s
arg_pat = re.compile(r"-+[^=\s]+[=\s](.*)")
match = arg_pat.match(line)
if match:
    line = match.group(1)
...